astropy/modeling/tests/test_quantities_fitting.py

```diff
@@ -93,15 +90,27 @@ def test_fitting_missing_data_units():
     """
     Raise an error if the model has units but the data doesn't
     """
-    g_init = models.Gaussian1D(amplitude=1. * u.mJy,
-                               mean=3 * u.cm,
-                               stddev=2 * u.mm)
+    class UnorderedGaussian1D(models.Gaussian1D):
+        # Parameters are ordered differently here from Gaussian1D
+        # to ensure the order does not break functionality.
+        def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
+            return {'amplitude': outputs_unit['y'],
+                    'mean': inputs_unit['x'],
+                    'stddev': inputs_unit['x']}
+
+    g_init = UnorderedGaussian1D(amplitude=1. * u.mJy,
+                                 mean=3 * u.cm,
+                                 stddev=2 * u.mm)
     fit_g = fitting.LevMarLSQFitter()
 
+    # We define flux unit so that conversion fails at wavelength unit.
+    # This is because the order of parameter unit conversion seems to
+    # follow the order defined in _parameter_units_for_data_units method.
     with pytest.raises(UnitsError) as exc:
-        fit_g(g_init, [1, 2, 3], [4, 5, 6])
-    assert exc.value.args[0] == ("'cm' (length) and '' (dimensionless) are not "
-                                 "convertible")
+        fit_g(g_init, [1, 2, 3],
+              [4, 5, 6] * (u.erg / (u.s * u.cm * u.cm * u.Hz)))
+    assert exc.value.args[0] == ("'cm' (length) and '' (dimensionless) are "
+                                 "not convertible")
```

Comment:
I meant the test data can be kept the same in which case the expected error message should be changed (because it omes from a different parameter) but this works too. Thanks!